core/ledger/kvledger/kv_ledger_test.go

```diff
@@ -567,6 +573,363 @@ func TestLedgerWithCouchDbEnabledWithBinaryAndJSONData(t *testing.T) {
+	assert.NoError(t, err)
+	defer lgr.Close()
+	lgr.(*kvLedger).setBLTPolicyForPvtStore(btlPolicyForSampleData())
+
+	bcInfo, _ := lgr.GetBlockchainInfo()
+	assert.Equal(t, &common.BlockchainInfo{
+		Height: 1, CurrentBlockHash: gbHash, PreviousBlockHash: nil,
+	}, bcInfo)
+
+	kvlgr := lgr.(*kvLedger)
+
+	sampleData := sampleDataWithPvtdataForSelectiveTx(t, bg)
+	for _, sampleDatum := range sampleData {
+		assert.NoError(t, kvlgr.commitToPvtAndBlockStore(sampleDatum))
+	}
+
+	// block 2 has no pvt data
+	pvtdata, err := lgr.GetPvtDataByNum(2, nil)
+	assert.NoError(t, err)
+	assert.Nil(t, pvtdata)
+
+	// block 5 has no pvt data
+	pvtdata, err = lgr.GetPvtDataByNum(5, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(pvtdata))
+
+	// block 3 has pvt data for tx 3, 5 and 6. Though the tx 6
+	// is marked as invalid in the block, the pvtData should
+	// have been stored
+	pvtdata, err = lgr.GetPvtDataByNum(3, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, len(pvtdata))
+	assert.Equal(t, uint64(3), pvtdata[0].SeqInBlock)
+	assert.Equal(t, uint64(5), pvtdata[1].SeqInBlock)
+	assert.Equal(t, uint64(6), pvtdata[2].SeqInBlock)
+
+	// block 4 has pvt data for tx 4 and 6 only
+	pvtdata, err = lgr.GetPvtDataByNum(4, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(pvtdata))
+	assert.Equal(t, uint64(4), pvtdata[0].SeqInBlock)
+	assert.Equal(t, uint64(6), pvtdata[1].SeqInBlock)
+
+	blockAndPvtdata, err := lgr.GetPvtDataAndBlockByNum(3, nil)
+	assert.NoError(t, err)
+	assert.True(t, proto.Equal(sampleData[2].Block, blockAndPvtdata.Block))
+
+	blockAndPvtdata, err = lgr.GetPvtDataAndBlockByNum(4, nil)
+	assert.NoError(t, err)
+	assert.True(t, proto.Equal(sampleData[3].Block, blockAndPvtdata.Block))
+
+	// pvt data retrieval for block 3 with filter should return filtered pvtdata
+	filter := ledger.NewPvtNsCollFilter()
+	filter.Add("ns-1", "coll-1")
+	blockAndPvtdata, err = lgr.GetPvtDataAndBlockByNum(4, filter)
+	assert.NoError(t, err)
+	assert.Equal(t, sampleData[3].Block, blockAndPvtdata.Block)
+	// two transactions should be present
+	assert.Equal(t, 2, len(blockAndPvtdata.PvtData))
+	// both tran number 4 and 6 should have only one collection because of filter
+	assert.Equal(t, 1, len(blockAndPvtdata.PvtData[4].WriteSet.NsPvtRwset))
+	assert.Equal(t, 1, len(blockAndPvtdata.PvtData[6].WriteSet.NsPvtRwset))
+	// any other transaction entry should be nil
+	assert.Nil(t, blockAndPvtdata.PvtData[2])
+
+	// test missing data retrieval in the presence of invalid tx. Block 6 had
+	// missing data (for tx4 and tx5). Though tx5 was marked as invalid tx,
+	// both tx4 and tx5 missing data should be returned
+	expectedMissingDataInfo := make(ledger.MissingPvtDataInfo)
+	expectedMissingDataInfo.Add(6, 4, "ns-4", "coll-4")
+	expectedMissingDataInfo.Add(6, 5, "ns-5", "coll-5")
+	missingDataInfo, err := lgr.(*kvLedger).GetMissingPvtDataInfoForMostRecentBlocks(1)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedMissingDataInfo, missingDataInfo)
+}
+
```

Comment:
I did not notice the remaining tests migrated here... Particularly two tests that were testing the error paths. The exact logic in those tests is not of much importance, but mainly for testing the error path, can you verify that these error paths are getting tested in existing tests somewhere?

Reply:
Added `TestCommitToPvtAndBlockstoreError()`.